File: core-webserver/core-entities/src/main/java/com/example/starter/db/entity/UserEntity.java
package com.example.starter.db.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.util.UUID;
import org.jetbrains.annotations.Nullable;

@Entity
@Table(indexes = {
    @Index(name = "idx_user_tenant_username", columnList = "tenant_id,name", unique = true)})
public class UserEntity {

  @Id
  @GeneratedValue
  @Nullable
  public UUID id;
  @ManyToOne(fetch = FetchType.LAZY, optional = false)
  @JoinColumn(name = "tenant_id", nullable = false, updatable = false)
  public TenantEntity tenant;
  @Column(nullable = false)
  public String name;

  public UserEntity(TenantEntity tenant, String name) {
    this.tenant = tenant;
    this.name = name;
  }

  UserEntity() {
  }
}

File: core-webserver/core-webserver/src/main/kotlin/com/example/starter/db/entity/UserEntity.kt
package com.example.starter.db.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.Index
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.util.UUID

@Entity
@Table(
    indexes = [Index(name = "idx_user_tenant_username", columnList = "tenant_id,name", unique = true)]
)
class UserEntity(
    @Id
    @GeneratedValue
    var id: UUID?,
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "tenant_id", nullable = false, updatable = false)
    var tenant: TenantEntity?,
    @Column(nullable = false) var name: String?,
    @Suppress("UnusedPrivateMember") unused: Void?
) {
    constructor() : this(null, null, null, null)
    constructor(tenant: TenantEntity, name: String) : this(null, tenant, name, null)
}
